themes/digital.gov/layouts/news/card-news.html

```diff
@@ -0,0 +1,109 @@
+{{/* Display a news card for internal or external blog links.
+  External links use both source and source_url to set the logo and external link.
+*/}}
+
+{{/* Unique ID to link article to its title in screen readers. Turns "18f at ten" into "title-18f-at-ten" */}}
+{{ $titleID := (printf "%s%s" "title-" .Title) | urlize }}
+{{ $title := .Title }}
+{{ $kicker := .Params.kicker }}
+{{ $deck := .Params.deck }}
+{{ $summary := .Params.summary }}
+{{ $authors := .Params.authors }}
+{{/* source_url is an external link */}}
+{{ $externalLink := .Params.source_url }}
+{{ $internalLink := .Permalink }}
+
+{{/* Get the source to lookup the logo for an external image */}}
+{{ $source := .Params.source }}
+{{ $sourcePage := $.Site.GetPage "page" (print "source_" $source ) }}
+{{/* sourceName returns name of agency to display on page (i.e. "USA.gov", "The White House") */}}
+{{ $sourceName := $sourcePage.Params.name }}
+{{/* sourceLogo returns lowercase name of agency used for internal image mapping (i.e. "10x", "gsa", "usagov", "whitehouse") */}}
+{{ $sourceLogo := $sourcePage.Params.logo }}
+{{ $sourceLogoPath := (printf "logos/%s%s" $sourceLogo "-logo.png") }}
+
+{{/* Flag to display internal or external card styles */}}
+{{ $isExternal := false }}
+{{ if and $source $externalLink }}
+  {{ $isExternal = true }}
+{{ end }}
+
+
+<article
+  class="card {{ if $isExternal }}
+    card-news--external
+  {{ else }}
+    card-news--internal
+  {{ end }}card-linked"
```

Comment:
### Future enhancement

In the future we should look to avoid logic in classes. It's hard to read and impacts formatting in the output HTML.

**Output formatting**
![image](https://github.com/GSA/digitalgov.gov/assets/3385219/27989fd5-8881-4bdb-9820-718be7fdd588)

---

#### Potential workaround

We should consider a map/object that handles this.

**Pseudocode example**
```
$cardClasses = {
  "card",
  "card--linked",
  $isExternal ? "card--external" : "card--internal"
}

// Convert into a single string 
<article class="{{ $cardClasses }}">
```

Reply:
Refactored this logic to:

https://github.com/GSA/digitalgov.gov/blob/308392600b03c286e926d2ffce72fe1a84a15c2c/themes/digital.gov/layouts/news/card-news.html#L31

`card-linked` needs to come last and `card-news--internal` does not have styles but is a placeholder.